backtracking/sudoku_solve.cpp

```diff
@@ -1,91 +1,144 @@
+/**
+ * @file
+ * @brief [Sudoku Solver](https://en.wikipedia.org/wiki/Sudoku) algorithm.
+ *
+ * @details
+ * Sudoku (数独, sūdoku, digit-single) (/suːˈdoʊkuː/, /-ˈdɒk-/, /sə-/, originally called
+ * Number Place)[1] is a logic-based,[2][3] combinatorial[4] number-placement puzzle. 

```

Comment:
```suggestion
 * Number Place) is a logic-based, combinatorial number-placement puzzle. 

```

Reply:
Good point; what kind of description shall we add here?